packages/core/src/components/button/_button-group.scss

```diff
@@ -142,17 +142,25 @@ Styleguide button-group
   }
 
   &.#{$ns}-outlined {
-    > .#{$ns}-button {
+    .#{$ns}-button {
       @include pt-button-outlined();
     }
 
     &:not(.#{$ns}-vertical) {
+      > .#{$ns}-popover-target:not(:last-child) .#{$ns}-button,
       > .#{$ns}-button:not(:last-child) {
         border-right: none;
       }
     }
   }
 
+  &:not(.#{$ns}-vertical) {
+    > .#{$ns}-popover-target:not(:last-child) .#{$ns}-button.#{$ns}-outlined,
+    > .#{$ns}-button.#{$ns}-outlined:not(:last-child) {
+      border-right: none;
+    }
+  }
```

Comment:
Do we need this for vertical button groups too (with border-bottom)?

Reply:
Ah, good catch! Added in 79e3441e9138d848d591f34b974e630dfba5ecce